### PR TITLE
feat: participation sale progress

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Display users some information about the progress of their participation to a sale.

# PRs

- [x] gix-cmp [#160](https://github.com/dfinity/gix-components/pull/160)

# Changes

- add `Progress` step to `TransactionModal` and disable close if in progress
- new enum and component `SaleInProgress` to display progression
- set various steps progress in sale service
- close modal after sale without delay on error and with a small delay on success
- refactor / move few components around
- improve `snsTicketsStore` typescript declarations

# Screenshots

demo-dekstop
demo-mobile

### CODE REVIEW

File: frontend/src/lib/modals/accounts/NewTransaction/TransactionModal.svelte [hunk 1]
```diff
@@ -25,6 +25,8 @@
   ) => string | undefined = () => undefined;
   // TODO: Add transaction fee as a Token parameter https://dfinity.atlassian.net/browse/L2-990
 
+  const STEP_PROGRESS = "Progress" as const;
```
Comment (mstrasinskis) [inline frontend/src/lib/modals/accounts/NewTransaction/TransactionModal.svelte:28]: Why `as const`?

File: frontend/src/lib/services/sns-sale.services.ts [hunk 2]
```diff
@@ -709,10 +720,12 @@ export const participateInSnsSale = async ({
     if (err instanceof TxCreatedInFutureError) {
       const retryIn = SALE_PARTICIPATION_RETRY_SECONDS;
 
+      // TODO: todo remove this or await it to sync the progress and modal close
```
Comment (mstrasinskis) [inline frontend/src/lib/services/sns-sale.services.ts:723]: Jfyi, changes are already in my PR.

File: frontend/src/lib/stores/sns-tickets.store.ts [hunk 3]
```diff
@@ -11,13 +13,25 @@ export interface SnsTicketsStoreEntry {
   keepPolling: boolean;
 }
 
-export interface SnsTicketsStore {
-  // Root canister id is the key to identify the parameters for a specific project.
-  [rootCanisterId: string]: SnsTicketsStoreEntry | undefined;
+export type SnsTicketsStoreData = Record<
```
Comment (mstrasinskis) [inline frontend/src/lib/stores/sns-tickets.store.ts:16]: Thank you for improvement!

**Review by mstrasinskis — APPROVED** [review]
LGTM, thanks!

File: frontend/src/lib/modals/accounts/NewTransaction/TransactionModal.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/modals/accounts/NewTransaction/TransactionModal.svelte]: Indeed not needed here

**Pull request merged**